File: LeetCode/Java/EvenDigits.java
package LeetCode.Java;

// Question
// https://leetcode.com/problems/find-numbers-with-even-number-of-digits/

public class EvenDigits {

    static int[] arr = {555, 901, 482, 1771};
    static int count;

    public static void main(String[] args) {

        //System.out.println(findNumbers(arr));

        System.out.println(digit1(1771));

    }

    static int digits(int n) {

        // for negative numbers
        if (n < 0) {
            n = n * -1;
        }

        if (n == 0) {
            return 1;
        }
        int c = 0;
        while (n > 0) {
            c++;
            n = n / 10;
        }
        return c;
    }

    static int digit1(int num) {

        if (num < 0) {
            num = num * -1;
        }

        return (int) (Math.log10(num)) + 1;
    }

    static boolean even(int num) {
        int n = digit1(num);

        return n % 2 == 0;
    }

    static int findNumbers(int[] arr) {

        for (int i : arr) {
            if (even(i)) {
                count++;
            }
        }
        return count;
    }
}

File: LeetCode/Python/EvenDigits.py
import math

arr = [555, 91, 42, 1771]


def digit(num):
    if num < 0:
        num *= -1

    return int(math.log10(num)) + 1


def even(num):
    n = digit(num)

    return n % 2 == 0


def findNumbers(arr):
    count = 0
    for i in arr:
        if even(i):
            count += 1
    return count


print(findNumbers(arr))
